### PR TITLE
Add cudnn sum and test that exposes tf test failure for nvGPU

### PR DESCRIPTION
**Pull request by AyanmoI**
This change fixes tensorflow test failures

### CODE REVIEW

File: src/ngraph/runtime/gpu/gpu_emitter.cpp [hunk 1]
```diff
@@ -1563,6 +1563,11 @@ void runtime::gpu::GPU_Emitter::emit_Subtract(EMIT_ARGS)
 }
 
 void runtime::gpu::GPU_Emitter::emit_Sum(EMIT_ARGS)
+{
+    runtime::gpu::GPU_Emitter::emit_Sum_1(external_function, writer, node, args, out);
+}
+
+void runtime::gpu::GPU_Emitter::emit_Sum_0(EMIT_ARGS)
```
Comment (rkimballn1) [inline src/ngraph/runtime/gpu/gpu_emitter.cpp]: Comments for something ambiguously named are mandatory. What does this function do? Same comment for Sum_1
Also in comment, why this is better or worse or whatever than other functions. What does this function bring to the table?

File: test/backend_sum.in.cpp [hunk 2]
```diff
@@ -427,3 +427,30 @@ NGRAPH_TEST(${BACKEND_NAME}, sum_5d_to_scalar)
...
+    {
+        vector<float> tensor_val(shape_size(param->get_shape()));
+        rng.initialize(tensor_val);
+        args.push_back(tensor_val);
+    }
+
+    auto cpu_func = clone_function(*f);
+    auto bk_func = clone_function(*f);
+
+    auto cpu_results = execute(cpu_func, args, "CPU");
```
Comment (rkimballn1) [inline test/backend_sum.in.cpp]: This test is only valid if the CPU backend is built. Please make it conditional on that.

Comment (jbobba) [inline test/backend_sum.in.cpp]: why are we comparing to CPU and not interpreter?

File: src/ngraph/runtime/cpu/unit_test.manifest [hunk 3]
```diff
@@ -12,3 +12,4 @@ one_hot_vector_1_fp
 one_hot_vector_1_fp_nonint
 backwards_batch_norm_three_outputs
 backwards_maxpool_n2_c1_hw5_3x3_str2_max_pad1x2_2x3
+sum_stable_acc
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/unit_test.manifest]: Would prefer to have only failing tests here.

File: test/backend_sum.in.cpp (hunk 2 above)
Comment (AyanmoI) [reply test/backend_sum.in.cpp]: INTERPRETER is also producing incorrect results. The purpose of this test is address the issue of failures in tensorflow tests. Both GPU and INTERPRETER produced similar results which differs from tensorflow. CPU  passes the tf tests.

Comment (csullivan) [reply test/backend_sum.in.cpp]: @AyanmoI, if it is easy, we could quickly change the INTERPRETER to use the kahan sum and try comparing against it. Can you take a look?

Comment (aprocter) [reply test/backend_sum.in.cpp]: The current situation with respect to `Sum` is a bit messy. `INTERPRETER` is a naive sum, while `CPU` was switched to a Kahan sum some time ago. The behavior should not be inconsistent like this; if it's part of the spec that `Sum` needs to be (or for that matter is _allowed_ to be) a stable sum, we should write that into the spec.

@rkimballn1 has raised this issue a number of times, but we haven't gotten around to resolving it. I'd actually favor separate `Sum` and `StableSum` ops, and leaving it up to the bridge to select the correct one.

Comment (aprocter) [reply test/backend_sum.in.cpp]: Also plausible: the current `Sum`'s spec could be updated with some kind of stability guarantee, and we could add a `NaiveSum` if anybody actually wants it.

File: src/ngraph/runtime/gpu/gpu_emitter.cpp (hunk 1 above)
Comment (AyanmoI) [reply src/ngraph/runtime/gpu/gpu_emitter.cpp]: Done

Comment (AyanmoI) [reply src/ngraph/runtime/gpu/gpu_emitter.cpp]: Done

File: test/backend_sum.in.cpp (hunk 2 above)
Comment (AyanmoI) [reply test/backend_sum.in.cpp]: Done

Comment (AyanmoI) [reply test/backend_sum.in.cpp]: I just pushed a kahan sum implementation for INTERPRETER `Sum`

Comment (AyanmoI) [reply test/backend_sum.in.cpp]: Also @jbobba , changed to comparing with INTERPRETER rather than CPU

Comment (AyanmoI) [reply test/backend_sum.in.cpp]: Please suggest if there needs to be two separate `sum`s for INTERPRETER

File: test/backend_sum.in.cpp [hunk 4]
```diff
@@ -427,3 +427,38 @@ NGRAPH_TEST(${BACKEND_NAME}, sum_5d_to_scalar)
...
+    test::Uniform<float> rng(1000.0f, 1000.1f, 2112);
+    vector<vector<float>> args;
+    for (shared_ptr<op::Parameter> param : f->get_parameters())
+    {
+        vector<float> tensor_val(shape_size(param->get_shape()));
+        rng.initialize(tensor_val);
+        args.push_back(tensor_val);
+    }
+
+    auto cpu_func = clone_function(*f);
```
Comment (jbobba) [inline test/backend_sum.in.cpp]: minor nit. Change to ref_func and ref_results.

**Comment by AyanmoI** [comment]
Reference Issue: https://github.com/NervanaSystems/ngraph/issues/1989

**Pull request merged**